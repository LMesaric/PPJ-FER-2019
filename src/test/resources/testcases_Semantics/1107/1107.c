int a[2]={2};
char b[200]="   \"  \0  \n   '  ";

char x = '"';

int main(int a)
{ b[0]=(char)'\0'+-1;
return a;
}