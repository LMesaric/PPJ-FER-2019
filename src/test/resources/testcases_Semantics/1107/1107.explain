semantic: invalid implicit conversion from `int` to `char`
note: 'b [ 0 ] = ( char ) '\0' + - 1' on line 7
