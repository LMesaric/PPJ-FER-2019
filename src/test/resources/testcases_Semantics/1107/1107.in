<prijevodna_jedinica>
 <prijevodna_jedinica>
  <prijevodna_jedinica>
   <prijevodna_jedinica>
    <vanjska_deklaracija>
     <deklaracija>
      <ime_tipa>
       <specifikator_tipa>
        KR_INT 1 int
      <lista_init_deklaratora>
       <init_deklarator>
        <izravni_deklarator>
         IDN 1 a
         L_UGL_ZAGRADA 1 [
         BROJ 1 2
         D_UGL_ZAGRADA 1 ]
        OP_PRIDRUZI 1 =
        <inicijalizator>
         L_VIT_ZAGRADA 1 {
         <lista_izraza_pridruzivanja>
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        BROJ 1 2
         D_VIT_ZAGRADA 1 }
      TOCKAZAREZ 1 ;
   <vanjska_deklaracija>
    <deklaracija>
     <ime_tipa>
      <specifikator_tipa>
       KR_CHAR 2 char
     <lista_init_deklaratora>
      <init_deklarator>
       <izravni_deklarator>
        IDN 2 b
        L_UGL_ZAGRADA 2 [
        BROJ 2 200
        D_UGL_ZAGRADA 2 ]
       OP_PRIDRUZI 2 =
       <inicijalizator>
        <izraz_pridruzivanja>
         <log_ili_izraz>
          <log_i_izraz>
           <bin_ili_izraz>
            <bin_xili_izraz>
             <bin_i_izraz>
              <jednakosni_izraz>
               <odnosni_izraz>
                <aditivni_izraz>
                 <multiplikativni_izraz>
                  <cast_izraz>
                   <unarni_izraz>
                    <postfiks_izraz>
                     <primarni_izraz>
                      NIZ_ZNAKOVA 2 "   \"  \0  \n   '  "
     TOCKAZAREZ 2 ;
  <vanjska_deklaracija>
   <deklaracija>
    <ime_tipa>
     <specifikator_tipa>
      KR_CHAR 4 char
    <lista_init_deklaratora>
     <init_deklarator>
      <izravni_deklarator>
       IDN 4 x
      OP_PRIDRUZI 4 =
      <inicijalizator>
       <izraz_pridruzivanja>
        <log_ili_izraz>
         <log_i_izraz>
          <bin_ili_izraz>
           <bin_xili_izraz>
            <bin_i_izraz>
             <jednakosni_izraz>
              <odnosni_izraz>
               <aditivni_izraz>
                <multiplikativni_izraz>
                 <cast_izraz>
                  <unarni_izraz>
                   <postfiks_izraz>
                    <primarni_izraz>
                     ZNAK 4 '"'
    TOCKAZAREZ 4 ;
 <vanjska_deklaracija>
  <definicija_funkcije>
   <ime_tipa>
    <specifikator_tipa>
     KR_INT 6 int
   IDN 6 main
   L_ZAGRADA 6 (
   <lista_parametara>
    <deklaracija_parametra>
     <ime_tipa>
      <specifikator_tipa>
       KR_INT 6 int
     IDN 6 a
   D_ZAGRADA 6 )
   <slozena_naredba>
    L_VIT_ZAGRADA 7 {
    <lista_naredbi>
     <lista_naredbi>
      <naredba>
       <izraz_naredba>
        <izraz>
         <izraz_pridruzivanja>
          <postfiks_izraz>
           <postfiks_izraz>
            <primarni_izraz>
             IDN 7 b
           L_UGL_ZAGRADA 7 [
           <izraz>
            <izraz_pridruzivanja>
             <log_ili_izraz>
              <log_i_izraz>
               <bin_ili_izraz>
                <bin_xili_izraz>
                 <bin_i_izraz>
                  <jednakosni_izraz>
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          BROJ 7 0
           D_UGL_ZAGRADA 7 ]
          OP_PRIDRUZI 7 =
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <aditivni_izraz>
                    <multiplikativni_izraz>
                     <cast_izraz>
                      L_ZAGRADA 7 (
                      <ime_tipa>
                       <specifikator_tipa>
                        KR_CHAR 7 char
                      D_ZAGRADA 7 )
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          ZNAK 7 '\0'
                   PLUS 7 +
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      <unarni_operator>
                       MINUS 7 -
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          BROJ 7 1
        TOCKAZAREZ 7 ;
     <naredba>
      <naredba_skoka>
       KR_RETURN 8 return
       <izraz>
        <izraz_pridruzivanja>
         <log_ili_izraz>
          <log_i_izraz>
           <bin_ili_izraz>
            <bin_xili_izraz>
             <bin_i_izraz>
              <jednakosni_izraz>
               <odnosni_izraz>
                <aditivni_izraz>
                 <multiplikativni_izraz>
                  <cast_izraz>
                   <unarni_izraz>
                    <postfiks_izraz>
                     <primarni_izraz>
                      IDN 8 a
       TOCKAZAREZ 8 ;
    D_VIT_ZAGRADA 9 }
