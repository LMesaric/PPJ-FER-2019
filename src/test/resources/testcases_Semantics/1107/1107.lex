KR_INT 1 int
IDN 1 a
L_UGL_ZAGRADA 1 [
BROJ 1 2
D_UGL_ZAGRADA 1 ]
OP_PRIDRUZI 1 =
L_VIT_ZAGRADA 1 {
BROJ 1 2
D_VIT_ZAGRADA 1 }
TOCKAZAREZ 1 ;
KR_CHAR 2 char
IDN 2 b
L_UGL_ZAGRADA 2 [
BROJ 2 200
D_UGL_ZAGRADA 2 ]
OP_PRIDRUZI 2 =
NIZ_ZNAKOVA 2 "   \"  \0  \n   '  "
TOCKAZAREZ 2 ;
KR_CHAR 4 char
IDN 4 x
OP_PRIDRUZI 4 =
ZNAK 4 '"'
TOCKAZAREZ 4 ;
KR_INT 6 int
IDN 6 main
L_ZAGRADA 6 (
KR_INT 6 int
IDN 6 a
D_ZAGRADA 6 )
L_VIT_ZAGRADA 7 {
IDN 7 b
L_UGL_ZAGRADA 7 [
BROJ 7 0
D_UGL_ZAGRADA 7 ]
OP_PRIDRUZI 7 =
L_ZAGRADA 7 (
KR_CHAR 7 char
D_ZAGRADA 7 )
ZNAK 7 '\0'
PLUS 7 +
MINUS 7 -
BROJ 7 1
TOCKAZAREZ 7 ;
KR_RETURN 8 return
IDN 8 a
TOCKAZAREZ 8 ;
D_VIT_ZAGRADA 9 }
