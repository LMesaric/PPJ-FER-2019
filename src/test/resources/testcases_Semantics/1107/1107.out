<izraz_pridruzivanja> ::= <postfiks_izraz> OP_PRIDRUZI(7,=) <izraz_pridruzivanja>
